api/test/test_auth.py

```diff
@@ -206,21 +206,25 @@ def test_unauthed_response_headers(client):
         ("asc", "2022-01-01"),
     ],
 )
-def test_sorting_authed(
-    client, monkeypatch, test_auth_token_exchange, sort_dir, exp_indexed_on
-):
-    # Prevent DB lookup for ES results because DB is empty.
-    monkeypatch.setattr("api.views.image_views.ImageSerializer.needs_db", False)
-
-    time.sleep(1)
-    token = test_auth_token_exchange["access_token"]
-    query_params = {"unstable__sort_by": "indexed_on", "unstable__sort_dir": sort_dir}
-    res = client.get("/v1/images/", query_params, HTTP_AUTHORIZATION=f"Bearer {token}")
-    assert res.status_code == 200
+def test_sorting_authed(client, test_auth_token_exchange, sort_dir, exp_indexed_on):
+    with patch(
+        "api.views.image_views.ImageViewSet.get_db_results"
+    ) as mock_get_db_result:
+        mock_get_db_result.side_effect = lambda value: value
+        time.sleep(1)
+        token = test_auth_token_exchange["access_token"]
+        query_params = {
+            "unstable__sort_by": "indexed_on",
+            "unstable__sort_dir": sort_dir,
+        }
+        res = client.get(
+            "/v1/images/", query_params, HTTP_AUTHORIZATION=f"Bearer {token}"
+        )
```

Comment:
Only these lines, where the view code is being executed, need to be inside the `patch` context, everything else can be at at the outer level as it was before.

Reply:
Thank you for trying to push me to understand testing.